sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala

```diff
@@ -504,16 +504,29 @@ case class WholeStageCodegenRDD(@transient sc: SparkContext, var source: CodeAnd
       private[this] var iter = computeInternal(split, context)
 
       override def hasNext: Boolean = try {
-        iter.hasNext
-      } catch {
-        case _: ClassCastException =>
-          logInfo(s"ClassCastException, hence recompiling")
-          CodeGenerator.invalidate(source)
-          iter = computeInternal(split, context)
+        try {
           iter.hasNext
+        } catch {
+          case _: ClassCastException =>
+            logInfo(s"ClassCastException, hence recompiling")
+            CodeGenerator.invalidate(source)
+            iter = computeInternal(split, context)
+            iter.hasNext
+        }
+      } catch {
+        case e: Throwable =>
+          logError(s"\n${CodeFormatter.format(source)}")
+          throw e
+      }
+
+      override def next(): InternalRow = try {
+        iter.next()
+      } catch {
+        case e: Throwable =>
+          logError(s"\n${CodeFormatter.format(source)}")
+          throw e
       }
 
-      override def next(): InternalRow = iter.next()
     }
   }
 
```

Comment:
We had to switch on the janino debug. Where it is done ?

Reply:
https://github.com/SnappyDataInc/snappy-store/pull/295